### PR TITLE
Fail in validate sdk

### PR DESCRIPTION
**Pull request by chcosta**
Basically took https://github.com/dotnet/arcade/pull/1580 and validated it (very minor changes) so that we can get this merged.

### CODE REVIEW

**Comment by chcosta** [comment]
Created a few different failure builds and then here's the latest build (which should succeed) - https://dev.azure.com/dnceng/internal/_build/results?buildId=87840

File: eng/validate-sdk.ps1 [hunk 1]
```diff
@@ -37,9 +38,9 @@ function AddSourceToNugetConfig([string]$nugetConfigPath, [string]$source)
     $valueAttribute = $nugetConfig.CreateAttribute("value")
     $valueAttribute.Value = $source
     $newSource = $nugetConfig.CreateElement("add")
-    $newSource.Attributes.Append($keyAttribute)
-    $newSource.Attributes.Append($valueAttribute)
-    $packageSources.AppendChild($newSource)
+    $null = $newSource.Attributes.Append($keyAttribute)
+    $null = $newSource.Attributes.Append($valueAttribute)
+    $null = $packageSources.AppendChild($newSource)
```
Comment (jcagme) [inline eng/validate-sdk.ps1]: Not using `$null` anywhere?

Comment (chcosta) [reply eng/validate-sdk.ps1]: I don't understand the question.  Adding $null prevents this random output from the end of the Arcade build.

```
2019-02-05T17:20:18.6152405Z #text                                                             
2019-02-05T17:20:18.6155851Z -----                                                             
2019-02-05T17:20:18.6157876Z arcade-local                                                      
2019-02-05T17:20:18.6159823Z F:\vsagent\15\s\artifacts\validatesdk\packages\Release\NonShipping
```

https://dev.azure.com/dnceng/internal/_build/results?buildId=87776&view=logs&jobId=16ce49ee-8d0d-5a76-4958-88f78392b71f&taskId=206012c0-0eaa-56aa-8172-665f354177dc&lineStart=2814&lineEnd=2817&colStart=1&colEnd=96

It's not really required, but the random output bugged me so I added this.

Comment (chcosta) [reply eng/validate-sdk.ps1]: Ack, thanks, I'll create a PR

**Pull request merged**